csrc/backend_ops/tensorrt/batched_nms/allClassNMS.cu

```diff
@@ -85,6 +87,10 @@ __global__ void allClassNMS_kernel(const int num, const int num_classes,
   //__shared__ bool kept_bboxinfo_flag[CAFFE_CUDA_NUM_THREADS * TSIZE];
   extern __shared__ bool kept_bboxinfo_flag[];
 
+#if defined(__CUDA_ARCH__) && __CUDA_ARCH__ <=720
```

Comment:
can we move assert out of the kernel?

Reply:
Seems we can't from [this blog](https://www.rookiehpc.com/cuda/docs/__cuda_arch__.php). While in this [blog](https://llvm.org/docs/CompileCudaWithLLVM.html), `__CUDA_ARCH__` will be defined if NVCC is using clang as a preprocessor. Maybe we can do the check for some threads like the first thread?

Reply:
@lvhan028 any idea?

Reply:
I think we can try `static_assert`, which happens at compiling time

Reply:
I am not sure whether it brings in `Undefined Behavior`

Reply:
The documentation for `__CUDA_ARCH__` says that host code should not depend on it. However, we can query the compute capability of the device using `cudaDeviceGetAttribute`